addons/point_of_sale/static/src/app/store/models.js

```diff
@@ -1991,6 +2087,27 @@ export class Order extends PosModel {
         }
     }
     set_orderline_options(orderline, options) {
+        if (options.comboLines?.length) {
+            options.price = 0;
+            orderline.comboLines = [];
+            const sumPriceOfComboLines = options.comboLines.reduce(
+                (acc, comboLine) =>
+                    acc + this.pos.db.product_by_id[comboLine.product_id[0]].lst_price,
+                0
+            );
+            const comboReductionFactor = orderline.product.lst_price / sumPriceOfComboLines;
+            options.comboLines.forEach((comboLine) => {
+                const product = this.pos.db.product_by_id[comboLine.product_id[0]];
+                this.pos.addProductToCurrentOrder(product, {
+                    price: product.lst_price * comboReductionFactor + comboLine.combo_price,
+                    comboParent: orderline,
+                });
+            });
+        }
+        if (options.comboParent) {
+            orderline.comboParent = options.comboParent;
+            orderline.comboParent.comboLines.push(orderline);
+        }
```

Comment:
Take this logic outside of set_orderline_options. Put it before set_orderline_options is called. Also, make convert forEach into a normal for-loop then `await` the call to "addProductToCurrentOrder" -- to make sure that each component is added one after the other.

Reply:
is ok now?